tests/test_model.py

```diff
@@ -0,0 +1,636 @@
+        "charging_state": "Charging",
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.charger_level is None
+
+def test_TeslaFiVehicle_charger_level_none_unplugged():
+    """Test TeslaFiVehicle class charger_level property"""
+    vehicle_data = {
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.charger_level is None
+    
+def test_TeslaFiVehicle_is_defrosting_front():
+    """Test TeslaFiVehicle class is_defrosting property"""
+    vehicle_data = {
+        "is_front_defroster_on": "1"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.is_defrosting is True
+
+def test_TeslaFiVehicle_is_defrosting_rear():
+    """Test TeslaFiVehicle class is_defrosting property"""
+    vehicle_data = {
+        "is_front_defroster_on": "1"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.is_defrosting is True
+
+def test_TeslaFiVehicle_is_defrosting_mode():
+    """Test TeslaFiVehicle class is_defrosting property"""
+    vehicle_data = {
+        "defrost_mode": "1"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.is_defrosting is True
+
+def test_TeslaFiVehicle_is_defrosting_mode_False():
+    """Test TeslaFiVehicle class is_defrosting property"""
+    vehicle_data = {
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.is_defrosting is False
+
+def test_TeslaFiVehicle_tpms():
+    """Test TeslaFiVehicle class tpms property"""
+    vehicle_data = {
+        "tpms_front_left": "43.0",
+        "tpms_front_right": "42.5",
+        "tpms_rear_left": "41.5",
+        "tpms_rear_right": "44.0",
+        "pressure": "psi",
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert isinstance(vehicle.tpms, TeslaFiTirePressure)
+    assert vehicle.tpms.front_left == 43.0
+    assert vehicle.tpms.front_right == 42.5
+    assert vehicle.tpms.rear_left == 41.5
+    assert vehicle.tpms.rear_right == 44.0
+    assert vehicle.tpms.unit == "psi"
+
+def test_TeslaFiVehicle_tpms_none():
+    """Test TeslaFiVehicle class tpms property"""
+    vehicle_data = {
+        "tpms_front_left": None,
+        "tpms_front_right": None,
+        "tpms_rear_left": None,
+        "tpms_rear_right": None
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert isinstance(vehicle.tpms, TeslaFiTirePressure)
+    assert vehicle.tpms.front_left is None
+    assert vehicle.tpms.front_right is None
+    assert vehicle.tpms.rear_left is None
+    assert vehicle.tpms.rear_right is None
+    assert vehicle.tpms.unit == "psi"
+
+def test_TeslaFiTirePressure_constructor():
+    """Test TeslaFiTirePressure class"""
+    
+    tire_pressure = TeslaFiTirePressure(
+        front_left=43.0,
+        front_right=42.5,
+        rear_left=41.5,
+        rear_right=44.0,
+        unit="psi",
+    )
+    
+    assert isinstance(tire_pressure, TeslaFiTirePressure)
```

Comment:
Not sure that this test is productive. Since you just created the instance, it is obviously the type you just created. It's more effective at line 613, since it is asserting that the `TeslaFiVehicle.tpms` property got created correctly.